gadget/gadget.go

```diff
@@ -972,6 +1097,37 @@ func validatePartial(v *Volume) error {
 	return nil
 }
 
+func validateVolumeAssignments(volassigns []*VolumeAssignment, volumes map[string]*Volume) error {
+	// do basic validation for volume-assignments
+	for _, va := range volassigns {
+		if err := va.validate(volumes); err != nil {
+			return fmt.Errorf("volume-assignment variant %q: %v", va.Name, err)
+		}
+	}
+
+	// ensure no two variants are identical in terms of contents, as we
+	// can only do matching based on the first correct we find.
+	checker := make(map[string]string)
+	for _, vavariant := range volassigns {
+		var keys []string
+
+		// include the device path to ensure that the assignments
+		// atleast differ, as the same volumes may be reused for multiple
+		// devices
+		for name, da := range vavariant.Assignments {
+			keys = append(keys, name+da.Device)
```

Comment:
nitpicking, but you could have strange cases where the joined string is the same for different names/path. Maybe it would be fine to use `reflect.DeepEqual` on the maps as this comparison is needed very occasionally and there is no performance concern.

Reply:
Including the name of the variant name in the key will make this not happen afaict?

Reply:
I did something like this to avoid that scenario